### ISSUE TITLE
Symbol.hasInstance usage can cause _classCallCheck to incorrectly fail

### ISSUE THREAD

**Issue by jayphelps** [babel/babel#4698]
### Input Code

[REPL](https://babeljs.io/repl/#?babili=false&evaluate=true&lineWrap=false&presets=es2015&experimental=false&loose=false&spec=false&playground=true&code=class%20Foo%20%7B%20%20%0A%20%20static%20%5BSymbol.hasInstance%5D(instance)%20%7B%0A%20%20%20%20return%20false%3B%0A%20%20%7D%0A%7D%0A%0Aconst%20arr%20%3D%20new%20Foo()%3B%0Aconsole.log(arr%20instanceof%20Foo)%3B%20%2F%2F%20false&stage=0)

```js
class Foo {  
  static [Symbol.hasInstance](instance) {
    return false;
  }
}

const arr = new Foo();
console.log(arr instanceof Foo); // false
```

### Babel Configuration (.bablerc, package.json, cli command)

* preset-es2015

### Expected Behavior

Runs and logs `false`, because we have overidden the behavior of `instanceof Foo` to always do so via [`Symbol.hasInstance`](http://www.ecma-international.org/ecma-262/6.0/#sec-symbol.hasinstance). I tested in Chrome and Firefox with native ES2015 and both run it as expected, logging `false`.

### Current Behavior

`_classCallCheck` itself uses `instanceof` to confirm you used `new`, but since our `instanceof` check always returns false, it fails with `Cannot call a class as a function`.

Best I can tell, this will only happen when running the transpiled code in true ES2015 envs because the `Symbol.hasInstance` overriding how `instanceof` works is voodoo that isn't transpiled and the behavior itself not polyfillable. This is pretty meta, so my explanation may not be clear.

### Possible Solution

Not entirely sure. I could be mistaken, but I thought babel previously detected `Symbol.hasInstance` usage and then did some voodoo to detect it was later being checked with an `instanceof` to emulate the behavior. A quick check seems to suggest this was [5.2](https://github.com/babel/babel/blob/fc3912217ffdcf775ad1459cb173d19c868f3233/CHANGELOG.md#520) but with the major 6.0 rewrite it appears this ability was (un)intentionally lost--or I'm misremembering what it was able to do. I'm not sure how it would even be possible in every situation given that you'd have to track assignments `const Bar = Foo; foo instanceof Bar;` or I guess when someone uses `Symbol.hasInstance` you depot to always transform every `instanceof` check to some helper. Yikes.

This stuff is pretty advanced ES2015 that I'm willing to bet very very VERY few people even know exists, and less have used. I think there are much bigger fish to fry assuming this isn't a simple "oops" bug. I looked and couldn't find any existing source to handle this `Symbol.hasInstance` stuff anymore. Primarily documenting here so others who run into this in the future can collect their thoughts. Feel free to close with WONTFIX

### Context

Long story short, noticed it when trying to do some cleverness in a custom compiler runtime that targets JavaScript and I use babel to transform it to ES5. It really is a long story, I wouldn't worry about it.

### Your Environment

| software         | version
| ---------------- | -------
| Babel            |  6.17.0

**Comment by loganfsmyth** [comment]
Yeah, it's a tough one. We're tracking this in https://github.com/babel/babel/issues/4452 at the moment. :)

**Comment by jayphelps** [comment]
@loganfsmyth shit sorry. I swear I searched! 🐧

**Comment by loganfsmyth** [comment]
No problem :P
